rule.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto2";
+package dv360_excel_plugin;
+
+/* A message that represents a scheduled rule for manipulating DV360 entities.
+ */
+message Rule {
+  /* required
+   * The name of the rule, set by the user.
+   */
+  optional string name = 1;
+
+  /* required
+   * The id that uniquely identified each rule.
+   */
+  optional int32 id = 2;
+
+  // required
+  optional Action action = 3;
+
+  // required
+  optional Schedule schedule = 4;
+}
+
+/*
+ * A message representing the action applied to a DV360 entity.
+ */
+message Action {
+  enum Type {
+    UNSPECIFIED_TYPE = 0;
+    CHANGE_LINE_ITEM_STATUS = 1;
+  }
+
+  /* required
+   * The type of action.
+   */
+  optional Type type = 1;
+
+  /*
+   * The action parameters to change line items' status.
+   */
+  message ChangeLineItemStatusParams {
+    // Line items to which the action applies.
+    repeated int64 line_items = 1;
+
+    enum Status {
+      UNSPECIFIED_STATUS = 0;
+      ACTIVE = 1;
+      PAUSED = 2;
+    }
+
+    /* required
+     * The status of which the line item will be changed to.
+     * Can be set to either active or paused.
+     */
+    optional Status status = 2;
+  }
+
+  /* required
+   * The parameters used for each type of action.
+   */
+  oneof params {
+    ChangeLineItemStatusParams change_line_item_status_params = 2;
+  }
+}
+/* required
+ * A message representing the schedule the rule will be run.
+ */
+message Schedule {
+  enum Type {
+    UNSPECIFIED_SCHEDULE = 0;
+    REPEATING = 1;
+  }
+
+  /* required
+   * The rule can be scheduled for either repeating or one-time.
+   */
+  optional Type type = 1;
+
+  /*
+   * The parameters for a repeating rule.
+   */
+  message RepeatingParams {
+    /* required
+     * The cron expression representing the repeating schedule.
+     */
+    optional string cron_expression = 1;
+  }
+
+  /* required
+   * The parameters used for each type of scheduling.
+   */
+  oneof params {
+    RepeatingParams repeating_params = 2;
```

Comment:
why this is a 2 instead of 1?

Reply:
Field number 1 is used by ['type'](https://github.com/googleinterns/dv360-excel-plugin/blob/8aa67c43e9e073d0f2a8fa3b5cd0538eaedbe459/rule.proto#L77).